File: Python/limit-occurrences-in-sorted-array.py
# Time:  O(n)
# Space: O(1)

# array, inplace
class Solution(object):
    def limitOccurrences(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        i = 0
        for x in nums:
            if i-k >= 0 and nums[i-k] == x:
                continue
            nums[i] = x
            i += 1
        while len(nums) != i:
            nums.pop()
        return nums

File: C++/limit-occurrences-in-sorted-array.cpp
// Time:  O(n)
// Space: O(1)

// array, inplace
class Solution {
public:
    vector<int> limitOccurrences(vector<int>& nums, int k) {
        int i = 0;
        for (const auto& x : nums) {
            if (i - k >= 0 && nums[i - k] == x) {
                continue;
            }
            nums[i++] = x;
        }
        nums.resize(i);
        return nums;
    }
};
